cli/flox/doc/flox-activate.md

```diff
@@ -155,18 +155,29 @@ See [`manifest.toml(5)`](./manifest.toml.md) for more details on shell hooks.
 
 ## Variables used by `flox activate`
 
-`$FLOX_SHELL`
-:  When launching an interactive sub-shell, Flox launches the shell specified in
-   `$FLOX_SHELL` if it is set.
-   When printing a shell for sourcing in the current shell,
-   Flox will produce a script suitable for `$FLOX_SHELL` if it is set.
-
-`$SHELL`
-:  When launching an interactive sub-shell, Flox launches the shell specified in
-   `$SHELL` if it is set and `$FLOX_SHELL` is not set.
-   When printing a shell for sourcing in the current shell,
-   Flox will produce a script suitable for `$SHELL` if it is set
-   and `$FLOX_SHELL` is not set and Flox can't detect the parent shell.
+`$FLOX_SHELL`, `$FLOX_SHELL_NOCLOBBER`, `$SHELL`
+:  When activating an environment in "interactive" or "in-place" modes,
+   Flox will either launch an interactive sub-shell
+   or emit commands to configure an already-running (parent) shell.
+   In both of these cases Flox needs to know which shell to use,
+   and these variables are used to control the selection process.
+
+       * interactive: When launching an interactive sub-shell
+         Flox will invoke
+         the shell specified in `$FLOX_SHELL` if set
+         or fall back to invoke `$SHELL` by default.
+
+       * in-place: When performing an "in place" activation
+         Flox will attempt to detect its parent shell type unless overridden by
+         the `$FLOX_SHELL` variable,
+         and if it cannot detect its parent shell type then will
+         produce a script with syntax determined by `$SHELL`.
+
+    Flox also supports the `$FLOX_SHELL_NOCLOBBER` variable
+    which has the same effect as `$FLOX_SHELL`
+    except that it not cleared from the environment
+    in the subsequent activation.
+    It is used primarily for testing and is not intended as a user-facing feature.
```

Comment:
I don't think we need to document something and say "but you don't really need to know this"
```suggestion
```

Reply:
+1 for not documenting at all if the only purpose is our testing.

Reply:
As a rule any variable starting with FLOX ought to be documented. If y'all really don't want this to be a user-facing feature then we should rename this to _FLOX_*.

Reply:
`_FLOX_SHELL_…` makes sense 👍🏻

Reply:
OK will change to _FLOX_SHELL_FORCE